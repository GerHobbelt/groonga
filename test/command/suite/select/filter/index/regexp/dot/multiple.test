table_create Memos TABLE_NO_KEY
column_create Memos content COLUMN_SCALAR Text

table_create RegexpTokens TABLE_PAT_KEY ShortText \
  --default_tokenizer TokenRegexp
column_create RegexpTokens memos_content COLUMN_INDEX|WITH_POSITION \
  Memos content

load --table Memos
[
{"content": "alphabet: abcXYdeXYZfXg"},
{"content": "space:    abc  de   f g"},
{"content": "not match: abcXYde   fg"}
]

#@add-important-log-levels info
log_level --level info
#@collect-query-log true
select Memos \
  --filter 'content @~ "abc..de...f.g"' \
  --output_columns 'content, _score'
#@collect-query-log false
log_level --level notice
#@remove-important-log-levels info
