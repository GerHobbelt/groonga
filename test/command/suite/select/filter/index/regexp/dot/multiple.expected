table_create Memos TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Memos content COLUMN_SCALAR Text
[[0,0.0,0.0],true]
table_create RegexpTokens TABLE_PAT_KEY ShortText   --default_tokenizer TokenRegexp
[[0,0.0,0.0],true]
column_create RegexpTokens memos_content COLUMN_INDEX|WITH_POSITION   Memos content
[[0,0.0,0.0],true]
load --table Memos
[
{"content": "alphabet: abcXYdeXYZfXg"},
{"content": "space:    abc  de   f g"},
{"content": "not match: abcXYde   fg"}
]
[[0,0.0,0.0],3]
log_level --level info
[[0,0.0,0.0],true]
select Memos   --filter 'content @~ "abc..de...f.g"'   --output_columns 'content, _score'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        2
      ],
      [
        [
          "content",
          "Text"
        ],
        [
          "_score",
          "Int32"
        ]
      ],
      [
        "alphabet: abcXYdeXYZfXg",
        1
      ],
      [
        "space:    abc  de   f g",
        1
      ]
    ]
  ]
]
#|i| [object][search][index][key][regexp] <RegexpTokens.memos_content>
#|i| grn_ii_sel > (abc..de...f.g)
#|i| [ii][select][cursor][open] n=2 <abc>
#|i| [ii][select][cursor][open] n=1 <de>
#|i| [ii][select][cursor][open] n=1 <f>
#|i| [ii][select][cursor][open] n=1 <g>
#|i| exact: 2
#|i| hits=2
#>select --filter "content @~ \"abc..de...f.g\"" --output_columns "content, _score" --table "Memos"
#:000000000000000 filter(2): Memos.content "abc..de...f.g" regexp
#:000000000000000 select(2)
#:000000000000000 output(2)
#:000000000000000 send(0)
#<000000000000000 rc=0
log_level --level notice
[[0,0.0,0.0],true]
